insights/tests/client/connection/test_test_connection.py

```diff
@@ -0,0 +1,175 @@
+def test_test_connection_test_urls_success(test_urls, insights_connection):
+    """The connection test performs several API calls in case of no error."""
+    insights_connection.test_connection()
+    assert len(test_urls.mock_calls) > 1
+
+
+@parametrize_exceptions
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_test_urls_fail(test_urls, exception, insights_connection):
+    """The connection test is stopped after the first API call failure."""
+    test_urls.side_effect = exception
+
+    insights_connection.test_connection()
+    test_urls.assert_called_once()
+
+
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_error_log_no_catch(test_urls, insights_connection, caplog):
+    """The connection test doesn't log any ERROR in case of an unknown exception."""
+    test_urls.side_effect = Exception
+
+    with caplog.at_level(ERROR, logger="insights.client.connection"):
+        try:
+            insights_connection.test_connection()
+        except Exception:
+            pass
+
+    assert not caplog.record_tuples
+
+
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_error_log_success(test_urls, insights_connection, caplog):
+    """The connection test doesn't log any ERROR if all API calls succeed."""
+    with caplog.at_level(ERROR, logger="insights.client.connection"):
+        insights_connection.test_connection()
+
+    assert not caplog.record_tuples
+
+
+@parametrize_exceptions
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_error_log_fail(
+    test_urls, exception, insights_connection, caplog
+):
+    """An error message is logged on the ERROR level."""
+    test_urls.side_effect = exception
+
+    with caplog.at_level(ERROR, logger="insights.client.connection"):
+        insights_connection.test_connection()
+
+    assert caplog.record_tuples == [
+        (
+            "insights.client.connection",
+            ERROR,
+            "Connectivity test failed! Please check your network configuration",
+        )
+    ]
+
+
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_print_no_catch(test_urls, insights_connection, capsys):
+    """The connection test doesn’t print anything in case of an unknown exception."""
+    test_urls.side_effect = Exception
+
+    try:
+        insights_connection.test_connection()
+    except Exception:
+        pass
+
+    out, err = capsys.readouterr()
+    assert not out
+    assert not err
+
+
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_print_success(test_urls, insights_connection, capsys):
+    """The connection test prints a message pointing to a log file if all API calls succeed."""
+    insights_connection.test_connection()
+
+    out, err = capsys.readouterr()
+    assert out == "See %s for more details.\n" % (LOGGING_FILE,)
```

Comment:
Could you please use `.format()`-style formatting instead? This is harder to read.

Non-blocking comment.